### DISCUSSION TITLE
Dinit output on boot to Serial Console

### DISCUSSION THREAD

**Discussion by fictitiousexistence** [davmac314/dinit#422] [Q&A] [answered]
Hi! 
Is there something else I need to do to make dinit startup output to both the screen and serial on boot?

In grub I have the following appended to the linux kernel line:
`console=ttyS0,115200n8 console=tty0`
This properly shows the output of the kernel when I'm connected to my serial port.

And I have a dinit service for ttyS0 that works great after boot. 

But I can't seem to see the dinit startup process over serial connection. 

I see the kernel output and then a pause while dinit is loading and then the dinit ttyS0 service kicks in and I get a login prompt.

Thanks!

**Comment by davmac314** [comment, marked as answer]
> Is there something else I need to do to make dinit startup output to both the screen and serial on boot?

This can't be done since the kernel doesn't support it. Dinit outputs to the console device (`/dev/console`) and the kernel outputs that to only a single device, see: https://tldp.org/HOWTO/Remote-Serial-Console-HOWTO/configure-kernel.html

From there:

> The last console is the one Linux uses as the `/dev/console` device

> Kernel messages will appear on both the first virtual terminal and the serial port. Messages from the init system and the system logger will appear only on the first serial port. This can be slightly confusing when looking at the attached monitor: the machine will appear to boot and then hang. Don't panic, the init system has started but is now printing messages to the serial port but is printing nothing to the screen. If a getty has been configured then a login: prompt will eventually appear on the attached monitor.

(That second one is describing the opposite setup to you, i.e. the serial tty listed after the virtual terminal).

**Reply by fictitiousexistence** [reply]
Thanks so much, that's the info I was looking for. I really care more about connecting via console then via a display so I changed `console=ttyS0,115200n8 console=tty0` to `console=tty0 console=ttyS0,115200n8`
